### PR TITLE
Implement namespace service

### PR DESCRIPTION
**Pull request by sshah-wework**
Namespace Service implementation

### CODE REVIEW

File: src/main/java/marquez/core/services/NamespaceService.java [hunk 1]
```diff
@@ -1,30 +1,66 @@
...
+  public NamespaceService(NamespaceDAO namespaceDAO) {
+    this.namespaceDAO = namespaceDAO;
+  }
+
+  public Namespace create(Namespace namespace) throws UnexpectedException {
+    try {
+      Namespace newNamespace =
+          new Namespace(
+              UUID.randomUUID(),
+              namespace.getName().toLowerCase(),
```
Comment (wslulciuc) [inline src/main/java/marquez/core/services/NamespaceService.java]: I'd call `toLowerCase()` on the namespace name in the namespace constructor. That way we ensure consistent casing when instantiating a namespace.

File: src/test/java/marquez/api/NamespaceIntegrationTest.java [hunk 2]
```diff
@@ -33,6 +33,26 @@ public void testCreateNamespace() {
...
+        .put(Entity.json(createNamespaceRequest));
+    Response res =
+        APP.client()
+            .target(URI.create("http://localhost:" + APP.getLocalPort()))
+            .path("/namespaces/" + NAMESPACE_NAME)
+            .request(MediaType.APPLICATION_JSON)
+            .put(Entity.json(createNamespaceRequest));
+    CreateNamespaceResponse responseBody = res.readEntity(CreateNamespaceResponse.class);
+    assertEquals(Response.Status.OK.getStatusCode(), res.getStatus());
+    assertThat(responseBody.getNamespace().getCreatedAt()).isAfter(START_TIME);
```
Comment (wslulciuc) [inline src/test/java/marquez/api/NamespaceIntegrationTest.java]: We'll want to discuss which testing framework we'd like to use for consistency across tests. I think there's a lot we can do with the simplicity of `org.junit.Assert` :)

File: src/test/java/marquez/core/services/NamespaceServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    List<Namespace> namespacesFound = namespaceService.listNamespaces();
+    assertEquals(0, namespacesFound.size());
+  }
+
+  @Test(expected = UnexpectedException.class)
+  public void testListNamespaces_findAllException() throws UnexpectedException {
+    doThrow(UnableToExecuteStatementException.class).when(namespaceDAO).findAll();
+    namespaceService.listNamespaces();
+  }
+}
```
Comment (wslulciuc) [inline src/test/java/marquez/core/services/NamespaceServiceTest.java]: 💯

File: src/main/java/marquez/core/services/NamespaceService.java (hunk 1 above)
Comment (wslulciuc) [reply src/main/java/marquez/core/services/NamespaceService.java]: Sure! In the mapper also works.

Comment (sshah-wework) [reply src/main/java/marquez/core/services/NamespaceService.java]: moved into a mapper

**Pull request merged**